csi/volumeutils.py

```diff
@@ -1124,6 +1124,54 @@ def mount_glusterfs(volume, mountpoint, storage_options="", is_client=False):
     else:
         volname = volume["name"]
 
+    if volume['type'] == 'External':
```

Comment:
- IMO moving this check here is the right thing to do
- it calls for some extra changes as well, from existing code perspective there are some places where `mount_glusterfs_with_host` is directly invoked and those will fail due to this refactor
```
-> rg mount_glusterfs_with_host
csi/volumeutils.py
1131:            mount_glusterfs_with_host(volname,
1262:def mount_glusterfs_with_host(volname, mountpoint, hosts, options=None, is_client=False):
1358:    mount_glusterfs_with_host(hvol['g_volname'], mntdir, hvol['g_host'], hvol['g_options'])

csi/nodeserver.py
10:from volumeutils import (mount_glusterfs, mount_glusterfs_with_host,
87:                mount_glusterfs_with_host(gvolname, request.target_path, gserver, options, True)
```
- referring above, pls make changes in `csi/volumeutils.py:1131` and `csi/nodeserver.py:87` to use `mount_glusterfs` rather than the other method, I hope this makes sense.

Reply:
I've checked all calls to `mount_glusterfs_with_host`:
1. `csi/volumeutils.py:1131` => not necessary to change as this is within `mount_glusterfs´ function
2. `csi/nodeserver.py:87`: I'll tackle this one, I just need to check the best way to do it
3. `csi/volumeutils.py:1358`: I guess this one needs to be updated as well

Reply:
yes, I meant 1358 in place of 1131, thanks for checking.